www/app/api/chat/route.ts

```diff
@@ -204,10 +210,14 @@ export async function POST(req: NextRequest) {
     if (!stream) {
       throw new Error('Failed to get stream');
     }
+    // Handle the stream response correctly
+    if (stream instanceof Response) {
+      return stream;
+    }
 
     console.log('Got the Stream');
 
-    return new NextResponse(stream.body, {
+    return new NextResponse(stream, {
```

Comment:
Why this change?

Reply:
Returning the 402 status code (payment required) in the route required changing the logic for response handling.